File: AD-HOC/1940/main.py
def get_total(j, scores):
    total = [0] * j
    for i, score in enumerate(scores):
        total[i % j] += score
    return total


def get_winner(total):
    max_score = max(total)
    return len(total) - total[::-1].index(max_score)


j = int(input().split()[0])
scores = list(map(int, input().split()))
total = get_total(j, scores)
print(get_winner(total))

File: AD-HOC/1940/main.dart
import 'dart:io';
import 'dart:math';

List<int> getTotal(int j, List<int> scores) {
  List<int> total = List.filled(j, 0);
  for (var (i, score) in scores.indexed) {
    total[i % j] += score;
  }
  return total;
}

int getWinner(List<int> total) {
  int maxScore = total.reduce(max);
  return total.lastIndexOf(maxScore) + 1;
}

void main() {
  int j = int.parse(stdin.readLineSync()!.split(' ').first);
  List<int> scores = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  List<int> total = getTotal(j, scores);
  print(getWinner(total));
}
